internal/provisioner/plain/controllers/bundledeployment_controller.go

```diff
@@ -134,39 +183,14 @@ func (r *BundleDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			status = metav1.ConditionFalse
 			message = fmt.Sprintf("Failed to unpack the %s Bundle", bundle.GetName())
 		}
-		u.UpdateStatus(
-			updater.EnsureCondition(metav1.Condition{
-				Type:    rukpakv1alpha1.TypeHasValidBundle,
-				Status:  status,
-				Reason:  reason,
-				Message: message,
-			}),
-		)
-		return ctrl.Result{}, nil
-	}
-	u.UpdateStatus(
-		updater.EnsureCondition(metav1.Condition{
-			Type:    rukpakv1alpha1.TypeHasValidBundle,
-			Status:  metav1.ConditionTrue,
-			Reason:  rukpakv1alpha1.ReasonUnpackSuccessful,
-			Message: fmt.Sprintf("Successfully unpacked the %s Bundle", bundle.GetName()),
-		}))
-
-	desiredObjects, err := r.loadBundle(ctx, bundle, bd.GetName())
-	if err != nil {
-		u.UpdateStatus(
-			updater.EnsureCondition(metav1.Condition{
-				Type:    rukpakv1alpha1.TypeHasValidBundle,
-				Status:  metav1.ConditionFalse,
-				Reason:  rukpakv1alpha1.ReasonBundleLoadFailed,
-				Message: err.Error(),
```

Comment:
We have pretty granular conditions here in case of errors, which I think is very helpful for the user. By refactoring to simply return an error in the reconciliation loop, we lose a lot of information. Could we refactor but keep these informative conditions?

Reply:
Thanks for review!  It seems that `git diff` got confused.  The new code is like [this](https://github.com/operator-framework/rukpak/blob/c010d2ba7aaa42afe72bff30cf30d0b7c6fffe82/internal/provisioner/plain/controllers/bundledeployment_controller.go#L133).  It is returning the err from the loadBundle as a return value of the reconciler.
```
        // Get the unpacked Bundle contents from the storage layer's cache
        desiredObjects, err := r.loadBundle(ctx, bundle, bd.GetName())
        if err != nil {
                updateStatusHasValidBundle(&u, metav1.ConditionFalse, rukpakv1alpha1.ReasonBundleLoadFailed, err.Error())
                return ctrl.Result{}, err
        }
``` 
The `updateStatusHasValidBundle` set the same status.